queries/javascript.core.scm

```diff
@@ -14,20 +17,33 @@
       field_definition
   )
 )
+
+;; `name` scope with `export`
 (export_statement
   (_
       name: (_) @name
   ) @dummy
+
+  ;; We have a special case for this one.  Note we don't need to list the other
+  ;; special cases from above because they can't be exported
   (#not-type? @dummy variable_declarator)
 ) @_.domain
 
+;; Special cases for `(let | const | var) foo = ...;` because the full statement
+;; is actually a grandparent of the `name` node, so we want the domain to include
+;; this full grandparent statement.
 (
   [
+    ;;!! (const | let) foo = ...;
```

Comment:
Note that the `(const | let)` and `...` make this not strictly Javascript, so doesn't fully implement the idea from https://github.com/cursorless-dev/cursorless/issues/1524#issuecomment-1594788592.  On the fence about whether we want to go all out and make these testable or just use it as a convenient notation

I wonder if in cases where it's not fully valid syntax we leave off the `!!` so that future auto testing impl doesn't pick these up 🤔.  Or maybe we leave it so that we're forced to clean these up if we go down that road

We could also leave off the `!!` and `!` altogether, but in some cases the clarity is helpful; see below